File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let words = s.split(' ');

    // for(let word of words){
    //     word = word.split('').reverse().join('');
    // }

    // Because In JavaScript, when you use a for...of loop with an array, 
    // the variable word becomes a copy of each element, and modifying it won't 
    // affect the original array.

     for (let i = 0; i < words.length; i++) {
        words[i] = words[i].split('').reverse().join('');
    }

    return words.join(' ');
};

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.cpp
class Solution {
private:
    vector<string> split(string s){
        vector<string> output;
        string word;
        for(int i=0; i<s.size(); ++i){
            
            if(s[i]==' '){
                output.push_back(word);
                word="";
                continue;
                
            }

            word+=s[i];

        }
        if(word!="")
          output.push_back(word);
        return output;
    }
public:
    string reverseWords(string s) {
        vector<string> splittedStr = split(s);
        for(auto a : splittedStr){
            cout<<a<<endl;
        }
        string ans= "";
        for(auto str : splittedStr){
            reverse(str.begin(), str.end());
            ans+=str;
            ans+=' ';
        }
        ans.pop_back();
        return ans;
    }
};